pkg/platform/runtime/setup/setup.go

```diff
@@ -229,21 +229,19 @@ func (s *Setup) updateArtifacts() ([]artifact.ArtifactID, error) {
 		}
 
 		// Retrieve environment definitions for artifact
-		envDef, err := as.EnvDef(unpackedDir)
+		envDef, err := as.PrepareEnvDef(tempUnpackedDir, s.store.InstallPath(), cnst)
 		if err != nil {
 			return errs.Wrap(err, "Could not collect env info for artifact")
 		}
 
-		// Expand environment definitions using constants
-		envDef = envDef.ExpandVariables(cnst)
-		err = envDef.ApplyFileTransforms(filepath.Join(unpackedDir, envDef.InstallDir), cnst)
+		err = envDef.ApplyFileTransforms(filepath.Join(tempUnpackedDir, envDef.InstallDir), cnst)
```

Comment:
What would happen if we removed tempdir from the equation and we just gave this the install path?

This still feels like it's not architected right. Something here is out of order..

Reply:
I agree it still doesn't feel right. The correct solution will likely require a larger changeset. That said we can use the install directory but only after we've moved the files. So now we are doing the file relocation after the files have been moved to the install path.